### PR TITLE
feat: [sc-10730] Fault tolerance for provisioning resources during FTUX

### PR DESCRIPTION
**Pull request by neurosnap**
Story details: https://app.shortcut.com/aptible/story/10730

### CODE REVIEW

File: src/deploy/database/index.ts [hunk 1]
```diff
@@ -273,7 +273,6 @@ export const provisionDatabase = thunks.create<
       dbId: `${dbCtx.json.data.id}`,
       containerSize: 1024,
       diskSize: 10,
-      status: "queued",
```
Comment (madhuravius) [inline src/deploy/database/index.ts:276]: just sanity check, was this dropped because this is on the return?

File: src/deploy/operation/index.ts [hunk 2]
```diff
@@ -37,6 +37,8 @@ export interface DeployOperationResponse {
   aborted: boolean;
   immediate: boolean;
   provisioned_iops: number;
+  disk_size: number;
+  container_size: number;
```
Comment (madhuravius) [inline src/deploy/operation/index.ts:41]: same as disk, if you choose to do it

File: src/deploy/database/index.ts (hunk 1 above)
Comment (neurosnap) [reply src/deploy/database/index.ts:276]: Yea we don't need these when submitting operations.

File: src/ui/pages/create-project-git.tsx [hunk 3]
```diff
@@ -1444,13 +1469,36 @@ export const CreateProjectGitStatusPage = () => {
...
+            changes, and then push the code to us to redeploy. This screen will
+            track your deployment status.
+          </p>
+          <p className="text-black-500 my-4">
+            You can also try to trigger a redeploy now if you think it was a
+            transient error that caused the deployment to fail.
+          </p>
+
+          <Button
+            onClick={() => redeploy(true)}
```
Comment (madhuravius) [inline src/ui/pages/create-project-git.tsx:1528]: minor nit, is you might want to set a timeout on attempting to redeploy so people don't accidentally double click it, so set disabled and unset it after 5-10s

File: src/deploy/operation/index.ts [hunk 4]
```diff
@@ -37,6 +37,8 @@ export interface DeployOperationResponse {
   aborted: boolean;
   immediate: boolean;
   provisioned_iops: number;
+  disk_size: number;
```
[the remark this answers is not in the document]
Comment (neurosnap) [reply src/deploy/operation/index.ts:40]: Let's save this for later.

File: src/ui/pages/create-project-git.tsx (hunk 3 above)
Comment (neurosnap) [reply src/ui/pages/create-project-git.tsx:1516]: Let's save this for testing.  The retry button will disappear almost immediately with its current function.

**Pull request merged**